[EDGE] Edge 2025 release updates

bbyreddy opened the pull request.

bbyreddy committed:
Version changes from 1018.0.x to 2025.0.x

bbyreddy committed:
Removed the --kube-apiserver-arg=admission-control=ValidatingAdmissionWebhook,MutatingAdmissionWebhook option from k3s install script

bbyreddy committed:
Version changes from 1018.0.x to 2025.0.x

bbyreddy committed:
Prerequisite steps for upgrading 1018 to 2025

BeateRixen commented on content/edge-kubernetes/manage-edge-bundle/version-upgrade.md:
```diff
@@ -10,9 +10,21 @@ Recreating update strategy is an all-or-nothing process that updates all aspects
 For this example, assume that Edge is deployed using the [c8yedge-sample.yaml](/files/edge-k8s/c8yedge-sample.yaml).
 
+### Upgrading Edge version 1018
+The Edge Operator version `{{< c8y-edge-current-version >}}` leverages the **admission webhooks** feature of Kubernetes to enhance validation and enforce default configurations for the Edge CR.
```
```suggestion
The Edge operator version `{{< c8y-edge-current-version >}}` leverages the **admission webhooks** feature of Kubernetes to enhance validation and enforce default configurations for the Edge CR.
```

bbyreddy replied:
Thank you, @BeateRixen

BeateRixen committed:
Update content/edge-kubernetes/manage-edge-bundle/version-upgrade.md

bbyreddy committed:
Version changes from 1018.0.x to 2025.0.x

bbyreddy committed:
Version changes from 1018.0.x to 2025.0.x

bbyreddy committed:
Version changes from 1018.0.x to 2025.0.x

bbyreddy committed:
Management tenant's domain name changes.

bbyreddy committed:
Management tenant's domain name changes.

bbyreddy committed:
Version changes from 1018.0.x to 2025.0.x

bbyreddy committed:
Prequisites and storage for Messaging Service

bbyreddy committed:
Prerequisite steps for upgrading 1018 to 2025

bbyreddy committed:
Prerequisites and storage for Messaging Service

bbyreddy committed:
Updated the validation and installation events to match the existing text

bbyreddy committed:
cumulocity-core service replaced with cumulocity-ontoplb service.

bbyreddy committed:
Databroker Agent Server microservice resource configuration updates.

bbyreddy committed:
Updated the "Data exchange using data broker" section to explain the operations sync and remove earlier unnecessary restrictions.

bbyreddy committed:
Messaging Service configuration description.

Commit:
Merge branch 'release/y2025' into edge/y2025-updates-first-cut

bbyreddy committed:
Apama metrics prefix updated to c8yedge_apama_ctrl_

bbyreddy committed:
Improvements as requested in CIE-5277

bbyreddy committed:
Messaging Service configuration description.

bbyreddy committed:
Aligned the script in the prerequisites with that in the Quick start guide.

rthrippleton commented on content/edge-kubernetes/manage-edge-bundle/version-upgrade.md:
```diff
@@ -10,9 +10,21 @@ Recreating update strategy is an all-or-nothing process that updates all aspects
 
 For this example, assume that Edge is deployed using the [c8yedge-sample.yaml](/files/edge-k8s/c8yedge-sample.yaml).
 
+### Upgrading Edge version 1018
```
```suggestion
### Upgrading from Edge version 10.18
```
(I think 1018 is purely an internal notation?)

bbyreddy replied:
We can be explicit, like 1018.0.x

rthrippleton replied:
Is that still correct though? Other bits of doc call it "10.18", specifically with the dot in there e.g. "Edge 10.18" at the top-left of our doc pages.

bbyreddy replied:
You are right. I will change it to "Upgrading from Edge version 10.18"

rthrippleton commented on content/edge-kubernetes/manage-edge-bundle/version-upgrade.md:
```diff
@@ -10,9 +10,21 @@ Recreating update strategy is an all-or-nothing process that updates all aspects
+- The [validating admission webhook](https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#validatingadmissionwebhook) ensures that any changes to the Edge CR comply with required constraints.
+- The [mutating admission webhook](https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#mutatingadmissionwebhook) automatically applies custom default values where needed.
+
+To enable admission webhooks, the Edge operator requires additional permissions. Before upgrading Edge to `{{< c8y-edge-current-version >}}`, run the following command to patch the `c8yedge-operator-manager-role` ClusterRole with the necessary permissions:
```
This implies that this is an optional step i.e. if you don't care about admission webhooks, ignore this step. I think this is a mandatory step though, right?

In which case, emphasise that, and put the "what you have to do" before the "why you have to do it" - to a user uninterested in these details, they just want to see "run this command so it works".

bbyreddy replied:
Rearranged the section to highlight what is required and explin why this setp is needed later.

rthrippleton commented on themes/c8ydocs/layouts/shortcodes/c8y-edge-current-version.html:
```diff
@@ -1 +1 @@
-{{- "1018" -}}
+{{- "2025" -}}
```
As if by accident, `content/edge/edge-databroker-bundle/databroker-edge.md` references this, so Edge appliance will start mentioning 2025! Needs fixing.

bbyreddy replied:
Actually I wanted to see if we can disable the Edge Appliance tile in the Early release Doc.

bbyreddy replied:
As discussed, I am moving the version individually into the Edge Appliance and Edge on K8s tiles and removing the version from the parent tile. Also had to hard code the version 1018 in the databroker-edge.md as the version with a DOT separator is needed in here.

rthrippleton commented on content/edge-kubernetes/installing-edge-on-k8-bundle/prerequisites.md:
```diff
@@ -25,35 +25,33 @@ Run the command below to install Kubernetes version 1.25.13:
+
+   printf "\e[32mSuccessfully installed k3s!\e[0m\n" && \
+
+   k3s crictl pull rancher/klipper-lb:v0.4.4 && \
```
How sure are you that this is necessary? I just did a k3s install of 1.25.13 (without those arguments), and all of them were already present:
```
root@rit-Latitude-7330:~# k3s crictl images
IMAGE                                        TAG                 IMAGE ID            SIZE
docker.io/rancher/klipper-lb                 v0.4.4              af74bd845c4a8       4.92MB
docker.io/rancher/local-path-provisioner     v0.0.24             b29384aeb4b13       14.9MB
docker.io/rancher/mirrored-coredns-coredns   1.10.1              ead0a4a53df89       16.2MB
docker.io/rancher/mirrored-metrics-server    v0.6.3              817bbe3f2e517       29.9MB
docker.io/rancher/mirrored-pause             3.6                 6270bb605e12e       301kB
```

bbyreddy replied:
One of the customers iTac, complained about this (K3s downloads klipper-lb, metrics-server and local-path provisioner images when the workloads requiring them are deployed) and in our tests we saw that these images were indeed downloaded at runtime.

rthrippleton commented on content/edge-kubernetes/k8s-troubleshooting-bundle/edge-k8s-does-not-show-as-device-on-cloud.md:
```diff
@@ -23,12 +23,10 @@ Status:
 ……
 ……
- Normal    Ready                        1m00s  cumulocityiotedge  installed successfully
+ Normal    Ready                        1m00s  cumulocityiotedge  {{< product-c8y-iot >}} Edge installation is complete, and it's now running version {{< c8y-edge-current-version >}}.0.0-XXXX
```
This is fine, but reminds me that we need to change the actual message to match this! Currently `Cumulocity IoT Edge installation is complete`

rthrippleton replied:
Actually, just working on this right now.

rthrippleton replied:
https://github.com/Cumulocity-IoT/edge-k8s-operator/pull/1033

bbyreddy replied:
Yes, this is one of the PRs, I wanted to create before the early release.

bbyreddy committed:
Addressed review comments.

bbyreddy committed:
moving the version individually into the Edge Appliance and Edge on K8s tiles and removing the version from the parent tile. Also had to hard code the version 1018 in the databroker-edge.md as the version with a DOT separator is needed in here.

rthrippleton commented on content/edge-kubernetes/k8-connecting-devices-to-edge-bundle/connecting-devices.md:
```diff
@@ -12,7 +12,7 @@ Additionally, Edge offers:
 * OPC UA protocol. OPC UA protocols support through the OPC UA device gateway and OPC UA management service.
 
-* Cloud Fieldbus functionality to collect data from fieldbus devices and remotely manage them. For example, Modbus protocol. With the release of {{< product-c8y-iot >}} Edge version {{< c8y-edge-current-version-alt >}}, we are announcing the deprecation of the {{< company-c8y >}} Linux Agent included in the {{< product-c8y-iot >}} Edge offering. For further details see the **Deprecation of {{< co ...
+* Cloud Fieldbus functionality to collect data from fieldbus devices and remotely manage them. For example, Modbus protocol. With the release of {{< product-c8y-iot >}} Edge version 10.18, we are announcing the deprecation of the {{< company-c8y >}} Linux Agent included in the {{< product-c8y-iot >}} Edge offering. For further details see the **Deprecation of {{< company-c8y >}} Linux Agent in {{ ...
```
FYI, that release notes URL is no longer accessible. In fact, I can't see release notes for Edge anywhere - https://cumulocity.com/docs/change-logs/?component=.component-edge-on-kubernetes%2C.component-edge-vm-appliance

bbyreddy committed:
Merge branch 'release/y2025' into edge/y2025-updates-first-cut

bbyreddy committed:
Update connecting-devices.md

bbyreddy committed:
Merge branch 'edge/y2025-updates-first-cut' of https://github.com/Cumulocity-IoT/c8y-docs into edge/y2025-updates-first-cut

bbyreddy committed:
Update connecting-devices.md

BeateRixen commented:
@bbyreddy Should I already do a language review or are you still on it?

bbyreddy commented:
> @bbyreddy Should I already do a language review or are you still on it?

@BeateRixen , yes you can start the language review. I will have to update the ticket numbers which I can do while you review. Thanks!

BeateRixen commented on content/edge-kubernetes/edge-custom-resource-definition-bundle/k8-edge-cloud-tenant.md:
```diff
@@ -6,9 +6,13 @@ layout: redirect
 Edge can be managed, configured, and monitored remotely through a {{< product-c8y-iot >}} cloud tenant. You can control and troubleshoot your Edge deployments remotely.
 
-To enable this, first register Edge as a device in the {{< product-c8y-iot >}} cloud tenant. You can register Edge by providing the {{< product-c8y-iot >}} cloud tenant URI, and optionally, TLS/SSL key and certificate chain with which Edge connects to cloud through MQTT protocol using a X.509 certificate for authentication. If you do not provide the TLS/SSL key and the certificate chain, the Edge ...
+To enable this, you must first register the Edge as a device within the cloud tenant. This registration process requires providing the {{< product-c8y-iot >}} cloud tenant URI, along with an optional TLS/SSL key and certificate chain. These credentials authenticate Edge when connecting to the cloud via the MQTT protocol using X.509 certificate-based authentication.
```
```suggestion
To enable this, you must first register Edge as a device within the cloud tenant. This registration process requires providing the {{< product-c8y-iot >}} cloud tenant URI, along with an optional TLS/SSL key and certificate chain. These credentials authenticate Edge when connecting to the cloud via the MQTT protocol using X.509 certificate-based authentication.
```

BeateRixen commented on content/edge-kubernetes/edge-custom-resource-definition-bundle/k8-edge-cloud-tenant.md:
```diff
@@ -17,7 +21,8 @@ kubectl get edge c8yedge -n c8yedge --output jsonpath='{.status.helpCommands.fet
 {{< /c8y-admon-info >}}
 
-Once registered, you can access your Edge remotely as a device, monitor its metrics, upgrade its version and collect diagnostic data remotely.
+Once registered, the Edge appears as a device in the cloud tenant. The device name is determined by the Common Name (CN) specified in the TLS/SSL certificate used for authentication. If you provided a custom TLS/SSL key and certificate, the device name matches the CN. However, if the Edge operator generated the TLS/SSL certificate internally, the device name defaults to the domain name of your Ed ...
```
```suggestion
Once registered, Edge appears as a device in the cloud tenant. The device name is determined by the Common Name (CN) specified in the TLS/SSL certificate used for authentication. If you provided a custom TLS/SSL key and certificate, the device name matches the CN. However, if the Edge operator generated the TLS/SSL certificate internally, the device name defaults to the domain name of your Edge.
```

BeateRixen commented on content/edge-kubernetes/edge-custom-resource-definition-bundle/k8-edge-microservices.md:
```diff
@@ -1,12 +1,12 @@
 |<div style="width:150px">Field</div>|Required|<div style="width:115px">Type</div>|Default|Description|
 |:---|:---|:---|:---|:---|
-|name|Yes|String|| The name of the {{< product-c8y-iot >}} microservice. The allowed values are apama-ctrl, smartrule, ssl-management-server, device-simulator, and opcua-mgmt-service
+|name|Yes|String|| The name of the {{< product-c8y-iot >}} microservice. The allowed values are apama-ctrl, smartrule, opcua-mgmt-service and databroker-agent-server
```
```suggestion
|name|Yes|String|| The name of the {{< product-c8y-iot >}} microservice. The allowed values are apama-ctrl, smartrule, opcua-mgmt-service and databroker-agent-server.
```

BeateRixen commented on content/edge-kubernetes/edge-custom-resource-definition-bundle/k8-edge-specification.md:
```diff
@@ -8,14 +8,15 @@ This section defines the Edge deployment's configurations.
 |licenseKey|Yes|String||Edge license key you received for the domain. <br>To request the license file for Edge, [contact product support](/additional-resources/contacting-support/)<br>In the email, you must include <p style="margin: 0; padding-left: 2em;">- Your company name, under which the license has been bought <p style="margin: 0; padding-left: 2em;">- The domain name (for example, myown.iot ...
 |company|Yes|String||Name of the "edge" tenant, for example, the company's name.
-|tlsSecretName| No|String|The Edge operator generates and assigns self-signed TLS/SSL private key and certificates.|Name of the Kubernetes secret containing the TLS/SSL private key and certificates for the domain name specified in the `spec.domain` field. This secret must contain two keys:<p style="margin: 0; padding-left: 2em;">- **tls.key:** TLS/SSL private key in the PEM format.<br>Generate a  ...
+|tlsSecretName| No|String|The Edge operator generates and assigns self-signed TLS/SSL private key and certificates.|Name of the Kubernetes secret containing the TLS/SSL private key and certificates for the domain name specified in the `spec.domain` field. This secret must contain two keys:<p style="margin: 0; padding-left: 2em;">- **tls.key:** TLS/SSL private key in the PEM format.<br>Generate a  ...
```
Should probably be "Edge tenant" instead of "edge tenant"?

bbyreddy replied:
Right it should be Edge tenant. I will change iit.

BeateRixen commented on content/edge-kubernetes/k8-edge-connecting-edge-to-cloud-bundle/data-exchange-using-data-broker.md:
```diff
@@ -8,37 +8,33 @@ sector:
-Data broker in Edge does not support synchronization of the [operations](/concepts/domain-model/#operations).
-{{< /c8y-admon-important >}}
+{{< c8y-admon-info >}}
+Devices, events, alarms, and measurements are forwarded from the Edge to the {{< product-c8y-iot >}} tenant account. In contrast, operations are forwarded from the {{< product-c8y-iot >}} tenant account to the Edge, but only for devices that exist in the {{< product-c8y-iot >}} tenant account due to forwarded data.
```
```suggestion
Devices, events, alarms, and measurements are forwarded from Edge to the {{< product-c8y-iot >}} tenant account. In contrast, operations are forwarded from the {{< product-c8y-iot >}} tenant account to Edge, but only for devices that exist in the {{< product-c8y-iot >}} tenant account due to forwarded data.
```

BeateRixen commented on content/edge-kubernetes/k8-edge-connecting-edge-to-cloud-bundle/data-exchange-using-data-broker.md:
```diff
@@ -8,37 +8,33 @@ sector:
 To create a data connector and upload the data to the tenant account, perform the following steps:
 
-1. In your Edge, log in to the "edge" tenant.
+1. Log in to the "edge" tenant. For more information on, see [To access Edge](/edge-kubernetes/installing-edge-on-k8/#to-access-cumulocity-iot-edge)
```
```suggestion
1. Log in to the "edge" tenant. For more information, see [To access Edge](/edge-kubernetes/installing-edge-on-k8/#to-access-cumulocity-iot-edge)
```

BeateRixen commented on content/edge-kubernetes/k8-edge-connecting-edge-to-cloud-bundle/k8-edge-register-edge-on-cloud.md:
```diff
@@ -4,15 +4,20 @@ title: Registering Edge in the cloud tenant
 ---
 
-To manage, configure, and monitor remotely through a {{< product-c8y-iot >}} cloud tenant, you must first register Edge as a device in the {{< product-c8y-iot >}} cloud tenant. You can register Edge by providing the {{< product-c8y-iot >}} cloud tenant URI, and optionally, TLS/SSL key and certificate chain with which Edge connects to cloud through MQTT protocol using an X.509 certificate for auth ...
+To remotely manage, configure, and monitor Edge through a {{< product-c8y-iot >}} cloud tenant, you must first register it as a device within the cloud tenant. This registration process requires providing the {{< product-c8y-iot >}} cloud tenant URI, along with an optional TLS/SSL key and certificate chain. These credentials authenticate Edge when connecting to the cloud via the MQTT protocol usi ...
```
```suggestion
To remotely manage, configure, and monitor Edge through a {{< product-c8y-iot >}} cloud tenant, you must first register it as a device within the cloud tenant. This registration process requires providing the {{< product-c8y-iot >}} cloud tenant URI, along with an optional TLS/SSL key and certificate chain. These credentials authenticate Edge when connecting to the cloud via the MQTT protocol using X.509 certificate-based authentication. You can configure the cloud tenant URI, TLS/SSL key, and certificate chain using the `spec.cloudTenant` field in the Edge CR. For more details, refer to [Cloud tenant](/edge-kubernetes/edge-custom-resource-definition/#k8-edge-cloud-tenant).
```

BeateRixen commented on content/edge-kubernetes/k8-edge-connecting-edge-to-cloud-bundle/k8-edge-register-edge-on-cloud.md:
```diff
@@ -4,15 +4,20 @@ title: Registering Edge in the cloud tenant
 {{< /c8y-admon-info >}}
 
-Once registered, the Edge deployment appears as a device named after the domain name of your Edge instance. For example, if you have configured your domain name as **myown.iot.com**, the Edge deployment appears as **myown.iot.com** .
+Once registered, the Edge appears as a device in the cloud tenant. The device name is determined by the Common Name (CN) specified in the TLS/SSL certificate used for authentication. If you provided a custom TLS/SSL key and certificate, the device name matches the CN. However, if the Edge operator generated the TLS/SSL certificate internally, the device name defaults to the domain name of your Ed ...
```
```suggestion
Once registered, Edge appears as a device in the cloud tenant. The device name is determined by the Common Name (CN) specified in the TLS/SSL certificate used for authentication. If you provided a custom TLS/SSL key and certificate, the device name matches the CN. However, if the Edge operator generated the TLS/SSL certificate internally, the device name defaults to the domain name of your Edge.
```

BeateRixen replied:
Or should we speak about "the Edge instance" here as you do below? We usually do not use Edge with an article.

bbyreddy replied:
I think here we can refer to Edge as "the Edge instance" same as below.

bbyreddy replied:
Changed to the Edge instance.

BeateRixen commented on content/edge-kubernetes/k8-edge-connecting-edge-to-cloud-bundle/k8-edge-register-edge-on-cloud.md:
```diff
@@ -4,15 +4,20 @@ title: Registering Edge in the cloud tenant
 
-Once registered, the Edge deployment appears as a device named after the domain name of your Edge instance. For example, if you have configured your domain name as **myown.iot.com**, the Edge deployment appears as **myown.iot.com** .
+Once registered, the Edge appears as a device in the cloud tenant. The device name is determined by the Common Name (CN) specified in the TLS/SSL certificate used for authentication. If you provided a custom TLS/SSL key and certificate, the device name matches the CN. However, if the Edge operator generated the TLS/SSL certificate internally, the device name defaults to the domain name of your Ed ...
+For example, if your Edge is configured with the domain **myown.iot.com**, it will appear as a device named **myown.iot.com** in the cloud tenant. Once registered, you can remotely access your Edge instance, monitor its metrics, perform version upgrades, and collect diagnostic data for troubleshooting and analysis.
```
"your Edge" also sounds weird to me. Better "your Edge instance"?

bbyreddy replied:
I agree, your Edge instance will also align with the Edge instance appearing in the cloud tenant.

bbyreddy replied:
Changed to the Edge instance.

BeateRixen commented on content/edge-kubernetes/edge-custom-resource-definition-bundle/k8-edge-cloud-tenant.md:
```diff
@@ -17,7 +21,8 @@ kubectl get edge c8yedge -n c8yedge --output jsonpath='{.status.helpCommands.fet
 
-Once registered, you can access your Edge remotely as a device, monitor its metrics, upgrade its version and collect diagnostic data remotely.
+Once registered, the Edge appears as a device in the cloud tenant. The device name is determined by the Common Name (CN) specified in the TLS/SSL certificate used for authentication. If you provided a custom TLS/SSL key and certificate, the device name matches the CN. However, if the Edge operator generated the TLS/SSL certificate internally, the device name defaults to the domain name of your Ed ...
+For example, if your Edge is configured with the domain **myown.iot.com**, it will appear as a device named **myown.iot.com** in the cloud tenant. Once registered, you can remotely access your Edge instance, monitor its metrics, perform version upgrades, and collect diagnostic data for troubleshooting and analysis.
```
See my comment below.

bbyreddy committed:
Update connecting-devices.md

bbyreddy committed:
Update connecting-devices.md

rthrippleton commented on content/edge-kubernetes/k8s-troubleshooting-bundle/edge-k8s-cr-status-install-update-loop-backoff.md:
````diff
@@ -27,14 +27,13 @@ Status:
  Normal  InstallLoopBackOff   1m00s  cumulocityiotedge  waiting for mongo server to be ready
 ```
 
+MongoDB requires a CPU that supports AVX instructions. Ensure that the CPU type of the Kubernetes node supports AVX instructions. Use the command `sudo lscpu` to check whether the CPU supports AVX instructions. 
````
Is `sudo` required anywhere? `lscpu` just reads `/proc/cpuinfo`, which can't be locked down afaik.

bbyreddy replied:
Not sure, had just copied this text from the prerequisites section. I will remove it and I think the user will use sudo if permissions error shows up.

rthrippleton commented on content/edge-kubernetes/k8-edge-introduction-bundle/versions.md:
```diff
@@ -6,9 +6,9 @@ sector:
-|[{{< product-c8y-iot >}} platform](/concepts)|{{< c8y-edge-current-version-alt >}}.0|
-|[Streaming Analytics](/streaming-analytics/introduction-analytics/)|10.18|
+|[{{< product-c8y-iot >}} platform](/concepts)|{{< c8y-edge-current-version >}}.0|
+|[Streaming Analytics](/streaming-analytics/introduction-analytics/)|25.339|
```
Especially considering the change in release versioning since 1018, I don't think it makes sense to separate out Streaming Analytics. It's just a component, like OPCUA version 1020.103.1 , or the messaging service 2025.0.2  (both as seen in the 2025 release). It's not special.

So I think it's time to collapse this just to say that Edge includes everything in the 2025 platform release.

bbyreddy replied:
We may have to be very explicit as by everything, people might misunderstand that it includes all the functionality/features etc..

rthrippleton replied:
Fair point. Let's ditch the table, simple statement that Edge _uses_ the Cumulocity platform version c8y-current-version, rather than _includes everything_ .

rthrippleton commented on content/edge-kubernetes/installing-edge-on-k8-bundle/configuring-storage.md:
```diff
@@ -35,6 +35,9 @@ The Edge operator requests three PVCs, as outlined in the table below. Each of t
 |75 GB|`mongod-data-edge-db-rs0-0`|Claimed by the MongoDB server to retain application data. The default size is 75 GB, but this value can be adjusted using the `spec.mongodb.resources.requests.storage` field in the Edge CR file.
 |10 GB|`microservices-registry-data`|Claimed by the private docker registry to store microservice images.
 |5 GB|`edge-logs`|Claimed by the Edge logging component to store the application and system logs.
+|10 GB|`pulsar-bookie-ledgers-pulsar-bookie-0`|Claimed by the Pulsar Bookkeeper Ledgers pod, which is deployed only when the Messaging Service is enabled.
```
Feel free to ignore this (I certainly don't care enough!), but you might want `GiB` in this whole table based on what you were doing yesterday.

rthrippleton replied:
... oh hold on, I got this the wrong way round. Never mind!

bbyreddy committed:
Addressing review comments

rthrippleton commented on content/edge-kubernetes/edge-custom-resource-definition-bundle/k8-edge-messaging-service.md:
```diff
@@ -0,0 +1,12 @@
+layout: redirect
+---
+
+Enables the installation of the {{< product-c8y-iot >}} Messaging Service, which is required for enabling the microservice-based data broker and Notifications 2.0. Enabling the Messaging Service requires additional system resources, including at least 2 extra CPU cores and 4 GB of RAM on the Kubernetes node. Additionally, three static Persistent Volumes (PVs) or a StorageClass with dynamic provis ...
```
The bit about PVs is slightly intimidating wording, implying I might need to take action. When in actual fact, by default this will just work; missing `storageClass` in the CR, just falls back to whatever k8s provides by default, normally local. Maybe copy the wording from the storageClassName definition (which is less intimidating), or just link back to it.

bbyreddy committed:
Apply suggestions from code review

Co-authored-by: Beate Rixen <90445236+BeateRixen@users.noreply.github.com>

bbyreddy committed:
Merge branch 'edge/y2025-updates-first-cut' of https://github.com/Cumulocity-IoT/c8y-docs into edge/y2025-updates-first-cut

bbyreddy committed:
removed sudo from lscpu command.

rthrippleton commented on content/edge-kubernetes/installing-edge-on-k8-bundle/prerequisites.md:
```diff
@@ -6,13 +6,13 @@ layout: redirect
 |Kubernetes|Version 1.25.x has been tested (with potential compatibility for subsequent versions) and is supported across the following platforms:<p style="margin: 0; padding-left: 2em;">- [Lightweight Kubernetes (K3s)](https://docs.k3s.io/installation). To enable the proper functioning of the Edge operator on K3s, you must install K3s with the following configuration options. For more informatio ...
 |Helm version 3.x|Refer to [Installing Helm](https://helm.sh/docs/intro/install/) for the installation instructions.|
-|Disk space|Three static Persistent Volumes (PV) or a StorageClass configured with dynamic provisioning to bind.<p style="margin: 0; padding-left: 2em;">- 75 GB for the Persistent Volume Claim (PVC) made for MongoDB (configurable through the Custom Resource).<p style="margin: 0; padding-left: 2em;">- 10 GB for the Persistent Volume Claim (PVC) made for the Private Registry to host custom microser ...
+|Disk space|Three static Persistent Volumes (PV) or a StorageClass configured with dynamic provisioning to bind.<p style="margin: 0; padding-left: 2em;">- 75 GB for the Persistent Volume Claim (PVC) made for MongoDB (configurable through the Custom Resource).<p style="margin: 0; padding-left: 2em;">- 10 GB for the Persistent Volume Claim (PVC) made for the Private Registry to host custom microser ...
```
Similar comment about 'intimidating wording' w.r.t. messaging provisioning. Best reword or link to existing wording, whatever you decide.

bbyreddy committed:
Update connecting-devices.md

rthrippleton commented on content/edge-kubernetes/edge-custom-resource-definition-bundle/k8-edge-cloud-tenant.md:
Minor style; this is a near-verbatim repetition of a bunch of text in the custom-resource-definition section. Would it be better to have the CRD section less verbose, and simply link in?

bbyreddy committed:
Merge branch 'release/y2025' into edge/y2025-updates-first-cut

bbyreddy committed:
[EDGE] y2024 release notes updates (#2896)

* Converted Edge release notes contents into Changelogs and also updated the Edge build artifact names to remove IOT from them.

* Included version dates for Edge 10.18

* Renamed Edge's build artifacts to "edge-operator" and "edge-appliance-vm" and Edge appliance component from "Edge VM Appliance" to "Edge Appliance VM"

* Changed the release dates in the date_settings.toml file for the 10.18.0.0 and 10.18.0.1 releases, so that these versions showup in the TOC of the Release Notes at the bottom of the list (before the 2024 releases)

* Release 2024 identical to 10.18 announcement is not required anymore as the Edge 10.18 release announcements showup under the respective versions.

* Fixed incorrect date format.

* Fixed release notes link.

* Update connecting-devices.md

* Deprecation announcement heading updated.

* Added tickets to the bug fixes and features where possible.

* Apply suggestions from code review

Co-authored-by: Beate Rixen <90445236+BeateRixen@users.noreply.github.com>

---------

Co-authored-by: Beate Rixen <90445236+BeateRixen@users.noreply.github.com>

bbyreddy committed:
[EDGE] y2024 release notes updates

bbyreddy committed:
Update connecting-devices.md

bbyreddy committed:
Merge branch 'release/y2025' into edge/y2025-updates-first-cut

bbyreddy committed:
[EDGE] Added version shortcodes for c8y-edge major, minor, fix and full. (#2906)

* Added version shortcodes for c8y-edge major, minor, fix and full.

* removed unnecessary -XXXX suffixes

* Removed unused edge version shortcode

bbyreddy committed:
Update install-operator-offline.md

bbyreddy committed:
Merge branch 'release/y2025' into edge/y2025-updates-first-cut

bbyreddy committed:
Adding messaging-service announcement to the release notes

bbyreddy committed:
updates

bbyreddy committed:
Update edge-on-kubernetes-2025.0.0-messaging-service.md

bbyreddy committed:
updates

bbyreddy committed:
updates

bbyreddy committed:
updates

bbyreddy committed:
Update k8-edge-specification.md

bbyreddy committed:
Update k8-edge-specification.md

bbyreddy committed:
Revert "Update k8-edge-specification.md"

This reverts commit a411fb39be69caf2e29b5a00a44c97a7fe89c87b.

bbyreddy committed:
updates

bbyreddy committed:
Update k8-edge-specification.md

bbyreddy committed:
Update k8-edge-specification.md

bbyreddy committed:
Update k8-edge-specification.md

BeateRixen commented on content/change-logs/edge/edge-appliance-vm-10-18-0-0-Deprecation-of-cumulocity-linux-agent-included-in-the-cumulocity-edge.md:
```diff
@@ -0,0 +1,24 @@
+---
+date: 2024-03-01
+title: Deprecation of Cumulocity Linux Agent included in the Cumulocity Edge
```
```suggestion
title: Deprecation of Cumulocity Linux agent included in Cumulocity Edge
```

BeateRixen committed:
Update content/change-logs/edge/edge-appliance-vm-10-18-0-0-Deprecation-of-cumulocity-linux-agent-included-in-the-cumulocity-edge.md

BeateRixen commented on content/change-logs/edge/edge-appliance-vm-10-18-0-0-Deprecation-of-cumulocity-linux-agent-included-in-the-cumulocity-edge.md:
```diff
@@ -0,0 +1,24 @@
+---
+date: 2024-03-01
+title: Deprecation of Cumulocity Linux agent included in Cumulocity Edge
```
```suggestion
title: Deprecation of Cumulocity Linux Agent included in Cumulocity Edge
```

BeateRixen committed:
Update content/change-logs/edge/edge-appliance-vm-10-18-0-0-Deprecation-of-cumulocity-linux-agent-included-in-the-cumulocity-edge.md

BeateRixen commented on content/change-logs/edge/edge-appliance-vm-10-18-0-0.md:
```diff
@@ -0,0 +1,19 @@
+ticket: CIE-4991
+version: 10.18.0.0
+---
+As CentOS 7 will reach its end-of-life on June 30th, 2024, we are transitioning our support to an alternative Operating System for Edge Appliance users. Starting with this release, the Edge Appliance VM is equipped with [Rocky Linux 8](https://rockylinux.org/). Since the Operating System is changed, the upgrade from earlier versions of Edge entails a side-by-side installation accompanied by data  ...
```
```suggestion
As CentOS 7 has reached its end-of-life on June 30th, 2024, we are transitioning our support to an alternative Operating System for Edge Appliance users. Starting with this release, the Edge Appliance VM is equipped with [Rocky Linux 8](https://rockylinux.org/). Since the Operating System is changed, the upgrade from earlier versions of Edge entails a side-by-side installation accompanied by data migration.
```

BeateRixen commented on content/change-logs/edge/edge-appliance-vm-10-18-0-0-Deprecation-of-cumulocity-linux-agent-included-in-the-cumulocity-edge.md:
```diff
@@ -0,0 +1,24 @@
+    label: edge-appliance-vm
+version: 10.18.0.0
+---
+With the release of {{< product-c8y-iot >}} Edge version 10.18, we are announcing the deprecation of the {{< product-c8y-iot >}} Linux Agent included in the {{< product-c8y-iot >}} Edge offering. Version 10.18 will mark the final release to feature the {{< product-c8y-iot >}} Linux Agent.
```
Since these are the release notes for 2025, is this entry still relevant? I also wonder about the version number. They show 10.18.0.x here and in the following release notes entries, Is this correct for 2025?

bbyreddy replied:
I copied all the 10.18 Edge Appliance release notes as the Edge Appliance will remain at 10.18. There are no futher releases planned for Edge Appliance. We plan to replace it with Edge Installer which will be release sometime in June 2025. Until then we need to keep Appliance VM 10.18 documentation.
Please suggest an alternate approach for this odd situation.

BeateRixen replied:
But we have the 10.18 Appliance VM documentation and release notes already in Release 2024, right? So at least for the release notes, I would probably only keep one entry which describes this situation and links to the 2024 Edge release notes?

bbyreddy replied:
Makes sense. I will remove all the 10.18 release notes entries from the 2025 release.
Also, what do you think we should be doing with Appliance 10.18? Do you think we should remove the Appliance tile from the Edge tiles, so we just have Edge on K8s tile in 2025 release docs? This is how the 2025.beta release doc looks: https://cumulocity-iot-c8y-docs-build-pr-2907.surge.sh/

BeateRixen replied:
If we remove it, it has consequences for a lot of links from other parts of the documentation. I would leave it but add one simple sentence of explanation, probably directly on the tile.

bbyreddy replied:
@BeateRixen , as discussed, I removed the 10.18 tickets from the change logs.

bbyreddy committed:
Update content/change-logs/edge/edge-appliance-vm-10-18-0-0.md

Co-authored-by: Beate Rixen <90445236+BeateRixen@users.noreply.github.com>

bbyreddy committed:
Update versions.md

bbyreddy committed:
Fixed broken links

bbyreddy committed:
Merge branch 'release/y2025' into edge/y2025-updates-first-cut

bbyreddy committed:
removed the 10.18 related entries from the release notes

bbyreddy committed:
Update quickstart.md

bbyreddy committed:
Merge branch 'release/y2025' into edge/y2025-updates-first-cut

bbyreddy committed:
Merge branch 'release/y2025' into edge/y2025-updates-first-cut

bbyreddy committed:
Merge branch 'release/y2025' into edge/y2025-updates-first-cut

rthrippleton committed:
Make PVs less intimidating, especially if you're using k3s

rthrippleton committed:
Remove a load of duplicated wording, cross-reference instead.

bbyreddy committed:
Minor changes to Edge Custom Resource link text

bbyreddy committed:
Update date_settings.toml

bbyreddy committed:
Simplified Disk space prerequisites.

bbyreddy committed:
Update prerequisites.md

bbyreddy committed:
Update prerequisites.md

bbyreddy committed:
Update prerequisites.md

bbyreddy committed:
Update prerequisites.md

BeateRixen commented on content/edge-kubernetes/edge-custom-resource-definition-bundle/k8-edge-messaging-service.md:
```diff
@@ -0,0 +1,13 @@
+layout: redirect
+---
+
+Enables the installation of the {{< product-c8y-iot >}} Messaging Service, which is required for enabling the microservice-based data broker and Notifications 2.0. Enabling the Messaging Service requires additional system resources, including at least 2 extra CPU cores and 4 GB of RAM on the Kubernetes node. Additionally, the Messaging Service will make use of three Persistent Volumes (PVs) for t ...
```
Enables the installation of the {{< product-c8y-iot >}} Messaging Service, which is required for enabling the microservice-based data broker and Notifications 2.0.

@bbyreddy This is a strange construction for a first sentence. What exactly enables the installation?

BeateRixen commented on content/edge-kubernetes/installing-edge-on-k8-bundle/prerequisites.md:
```diff
@@ -52,8 +51,7 @@ sudo sh -c '
 
-- Added `--disable=traefik` in the install command to disable Traefik to avoid port conflicts between Traefik and cumulocity-core service, as both are LoadBalancer type services which expose port 443.
-- Added `--kube-apiserver-arg=admission-control=ValidatingAdmissionWebhook,MutatingAdmissionWebhook` to enable admission controllers. The flag is set to enable the `ValidatingAdmissionWebhook` and `MutatingAdmissionWebhook` admission controllers, as Edge requires them. See [https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/](https://kubernetes.io/docs/reference/access- ...
+- Added `--disable=traefik` in the install command to disable Traefik to avoid port conflicts between Traefik and cumulocity-ontoplb service, as both are LoadBalancer type services which expose port 443.
```
```suggestion
- Added `--disable=traefik` in the install command to disable Traefik to avoid port conflicts between Traefik and cumulocity-ontoplb service, as both are load-balancer type services which expose port 443.
```

BeateRixen committed:
Update content/edge-kubernetes/installing-edge-on-k8-bundle/prerequisites.md

bbyreddy committed:
Addressed review comment.

bbyreddy committed:
Merge branch 'release/y2025' into edge/y2025-updates-first-cut

bbyreddy committed:
Update c8yedge_registry_sync-2025.0.0-py3-none-any.whl

bbyreddy committed:
Merge branch 'release/y2025' into edge/y2025-updates-first-cut

bbyreddy committed:
DB admin password is generated if not provided by the user, instead of using admin-pass.

bbyreddy committed:
Merge branch 'release/y2025' into edge/y2025-updates-first-cut

bbyreddy committed:
Update images

bbyreddy committed:
[EDGE CIE-5221] y2025 DataHub installation using the Edge Operator  (#3131)

* Don't hardcode use of /etc/hosts for datahub-<domain_name>
* Remove all doc related to Edge appliance + Datahub
* Edge appliance 10.18 + Datahub 2025 is no longer a valid combination 
However, we still have the Edge appliance 10.18 doc in here for Reasons,
so we're "vendoring" the old Datahub+Edge doc in here.
* Update the sector tag; left it in when copying from DataHub doc
* Removing the redundant datahub section under edge/

* DataHub for Edge on K8s updates
* Restructured and moved the DataHub content for the Edge Appliance VM into the Edge docs, so that the Edge content in the DataHub docs is all related to Edge on K8s.
---------

Co-authored-by: Richard Thrippleton <richard.thrippleton@cumulocity.com>
Co-authored-by: Beate Rixen <90445236+BeateRixen@users.noreply.github.com>

bbyreddy committed:
Merge branch 'release/y2025' into edge/y2025-updates-first-cut

bbyreddy committed:
Added API Change for the management tenant domain change

rthrippleton commented on content/change-logs/edge/edge-on-kubernetes-2025.0.0-management-tenant-domain.md:
```diff
@@ -0,0 +1,20 @@
+---
+date: 2025-03-31
+title: Changed {{< management-tenant >}} domain format for wildcard certificate Support 
```
Looks like Hugo isn't rendering shortcodes here
![image](https://github.com/user-attachments/assets/092bb063-2a04-4355-b77d-910ad99237e6)

bbyreddy replied:
Fixed https://github.com/Cumulocity-IoT/c8y-docs/pull/2880/commits/338a6ae7e85ad9bb80cc3b8ea7562363ddd7de84

bbyreddy committed:
Update edge-on-kubernetes-2025.0.0-management-tenant-domain.md

rthrippleton commented on content/change-logs/edge/edge-on-kubernetes-2025.0.0-management-tenant-domain.md:
```diff
@@ -0,0 +1,20 @@
+version: 2025.0.0
+---
+
+To support the use of a single wildcard certificate across multiple Edge instances, the format of the {{< management-tenant >}} domain has been changed. Instead of using `management.<edge-domain>`, it is now formatted as `management-<edge-domain>`. This change ensures better compatibility with wildcard certificates that match `*.yourdomain.com` but not nested subdomains like `management.edge1.you ...
```
I'd give an explicit example, because this description gets confusing if you don't notice subtle changes in punctuation :-)
```suggestion
To support the use of a single wildcard certificate across multiple Edge instances, the format of the {{< management-tenant >}} domain has been changed. Instead of using `management.<edge-domain>`, it is now formatted as `management-<edge-domain>`. This change ensures better compatibility with wildcard certificates that match `*.yourdomain.com`, which would match `management-edge1.yourdomain.com` but not nested subdomains like `management.edge1.yourdomain.com`.
```

rthrippleton commented on content/change-logs/edge/edge-on-kubernetes-2025.0.0-management-tenant-domain.md:
```diff
@@ -0,0 +1,20 @@
+---
+date: 2025-03-31
+title: Changed Management Tenant domain format for wildcard certificate support 
```
Pedantically, based on the existing shortcode:
```suggestion
title: Changed Management tenant domain format for wildcard certificate support 
```

bbyreddy committed:
Update content/change-logs/edge/edge-on-kubernetes-2025.0.0-management-tenant-domain.md

Co-authored-by: Richard Thrippleton <richard.thrippleton@softwareag.com>

bbyreddy committed:
Update content/change-logs/edge/edge-on-kubernetes-2025.0.0-management-tenant-domain.md

Co-authored-by: Richard Thrippleton <richard.thrippleton@softwareag.com>
